src/InsertionsClient/Api/Props/PropsVariableDeducer.cs

```diff
@@ -65,10 +65,25 @@ public PropsVariableDeducer(string feed, string? accessToken)
         /// <param name="swrFiles">Files, containing the variables.</param>
         /// <param name="deducedVariablesList">List of variables, value of which was found.</param>
         /// <param name="outcomeDetails">Detail string, explaining the outcome of the operation.</param>
-        /// <param name="maximumWaitSeconds">Maximum duration to wait for nuget downloads to complete.</param>
         /// <returns>True if operation succeeded. False otherwise</returns>
         public bool DeduceVariableValues(DefaultConfigUpdater defaultConfigUpdater, IEnumerable<PackageUpdateResult> packages,
-            SwrFile[] swrFiles, out List<PropsFileVariableReference> deducedVariablesList, out string outcomeDetails, int maximumWaitSeconds = -1)
+            SwrFile[] swrFiles, out List<PropsFileVariableReference> deducedVariablesList, out string outcomeDetails)
+        {
+            return DeduceVariableValues(defaultConfigUpdater, packages, swrFiles, out deducedVariablesList, out outcomeDetails, TimeSpan.FromMilliseconds(-1));
+        }
+
+        /// <summary>
+        /// Deduces the value of the variables in swr files
+        /// </summary>
+        /// <param name="defaultConfigUpdater"><see cref="DefaultConfigUpdater"/> that will provide extract locations for packages.</param>
+        /// <param name="packages">Nuget packages that should be downloaded and used.</param>
+        /// <param name="swrFiles">Files, containing the variables.</param>
+        /// <param name="deducedVariablesList">List of variables, value of which was found.</param>
+        /// <param name="outcomeDetails">Detail string, explaining the outcome of the operation.</param>
+        /// <param name="maximumWaitDuration">Maximum duration to wait for nuget downloads to complete.</param>
+        /// <returns>True if operation succeeded. False otherwise</returns>
+        public bool DeduceVariableValues(DefaultConfigUpdater defaultConfigUpdater, IEnumerable<PackageUpdateResult> packages,
+            SwrFile[] swrFiles, out List<PropsFileVariableReference> deducedVariablesList, out string outcomeDetails, TimeSpan maximumWaitDuration)
```

Comment:
nit: ```TimeSpan? maximumWaitDuration = null``` so the ugly -1 convention doesn't leak out of the method and there's no need for two overloads? Just a suggestion, not blocking.

Reply:
Makes sense. Does my change look like what you had in mind? Let's resolve this if yes.

Reply:
Yes, thanks, resolving!